File: geometry_basics_distance_between_points_in_2d/gbdbpi2.lua
-- [geometry basics distance between points in 2d]
-- [8 kyu]
-- [lua]
local solution = {}
function solution.distance_between_points(a, b)
local dx = b.x - a.x
local dy = b.y - a.y
return math.sqrt(dx*dx + dy*dy)
end
return solution

File: geometry_basics_distance_between_points_in_2d/gbdbpi2.cpp
// [geometry basics distance between points in 2d]
// [8 kyu]
// [c++]
#include <cmath>
double distance_between_two_points(const Point& a, const Point& b) 
{
double dx = b.x - a.x;
double dy = b.y - a.y;
double distance = std::hypot(dx, dy);
return distance;
}
